packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -152,3 +186,78 @@ async def publish(self, exchange_name: str, message: Message) -> None:
                 aio_pika.Message(message.encode()),
                 routing_key="",
             )
+
+    async def rpc_request(
+        self,
+        namespace: RPCNamespace,
+        method_name: str,
+        *,
+        timeout: Optional[PositiveInt] = 5,
+        **kwargs: dict[str, Any],
+    ) -> Any:
+        """
+        Call a remote registered `handler` by providing it's `namespace`, `method_name`
+        and list of expected arguments.
+
+        NOTE: `namespace` should always be composed via `get_namespace`
+        """
+
+        if not self._rpc:
+            raise RPCNotInitializedError()
+
+        namespaced_method_name = _get_namespaced_method_name(namespace, method_name)
+        try:
+            queue_expiration_timeout = timeout
+            awaitable = self._rpc.call(
+                namespaced_method_name,
+                expiration=queue_expiration_timeout,
+                kwargs=kwargs,
+            )
+            return await asyncio.wait_for(awaitable, timeout=timeout)
+        except MessageProcessError as e:
+            if e.args[0] == "Message has been returned":
+                raise RemoteMethodNotRegisteredError(
+                    method_name=namespaced_method_name, incoming_message=e.args[1]
+                ) from e
+            raise e
+
+    async def rpc_register_for(self, entries: dict[str, str], handler: Awaitable):
+        """
+        Bind a local `handler` to a `namespace` derived from the provided `entries`
+        dictionary.
+
+        NOTE: This is a helper enforce the pattern defined in `rpc_register`'s
+        docstring.
+        """
+        await self.rpc_register(
+            get_namespace(entries), method_name=handler.__name__, handler=handler
+        )
+
+    async def rpc_register(
+        self, namespace: RPCNamespace, method_name: str, handler: Awaitable
+    ) -> None:
+        """
+        Bind a local `handler` to a `namespace` and `method_name`.
+        The handler can be remotely called by providing the `namespace` and `method_name`
+
+        NOTE: method_name could be computed from the handler, but by design, it is
+        left to the caller to do so.
+        NOTE: `namespace` should always be composed via `get_namespace`
+        """
+
+        if self._rpc is None:
+            raise RPCNotInitializedError()
+
+        await self._rpc.register(
+            _get_namespaced_method_name(namespace, method_name),
+            handler,
+            auto_delete=True,
+        )
+
+    async def rpc_unregister(self, handler: Awaitable) -> None:
+        """Unbind a locally added `handler`"""
+
+        if self._rpc is None:
+            raise RPCNotInitializedError()
+
+        await self._rpc.unregister(handler)
```

Comment:
does the rpc not have its own registry? it seems they do it's a Proxy.

Reply:
Sorry I did not get the question.

Reply:
nevermind. I do not thing it brings anything.